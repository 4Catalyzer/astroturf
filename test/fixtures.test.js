const { basename, extname } = require('path');
const { fixtures, webpackRunFixture, babelRunFixture } = require('./helpers');

process.env.NODE_ENV = 'development';

const fixtureMatch = null; // /interpolations/;

describe('fixtures', () => {
  fixtures.forEach(fixture => {
    describe(basename(fixture, extname(fixture)), () => {
      webpackRunFixture(fixture, fixtureMatch && fixture.match(fixtureMatch));

      babelRunFixture(fixture, fixtureMatch && fixture.match(fixtureMatch));
    });
  });
});
