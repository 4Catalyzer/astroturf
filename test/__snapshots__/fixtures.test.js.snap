// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`css-literal-loader example js : Compiled JS 1`] = `
"import { css as less } from 'css-literal-loader/styled';
const SIZE = 75;

const styles =
/*#__PURE__*/
require(\\"./example-styles.less\\");"
`;

exports[`css-literal-loader example styles: example-styles.less 1`] = `
"
  .foo {
    color: red;
    width: 75px;
  }

  .bar {
    composes: foo;

    font-weight: bold;
    font-style: italic;
  }
"
`;

exports[`css-literal-loader hoist-import ButtonBase: hoist-import-ButtonBase.css 1`] = `
"@import '~./styles/mixins.scss';
.button-base {

  

  display: inline-flex;
  align-items: center;
  justify-content: center;
  border: 1px solid transparent;

}"
`;

exports[`css-literal-loader hoist-import js : Compiled JS 1`] = `
"import { styled } from 'css-literal-loader/styled';
const ButtonBase =
/*#__PURE__*/
styled('button', \\"ButtonBase\\", require(\\"./hoist-import-ButtonBase.css\\"), s => [s[\\"buttonBase\\"] || s[\\"button-base\\"]]);"
`;

exports[`css-literal-loader multiple-components FancyBox: multiple-components-FancyBox.css 1`] = `
".fancy-box {

  color: red;
  width: 75px;

  @media (min-width: 420px) {
		width: 96px;
		height: 96px;
	}

  

}
.fancy-box-variant-0 {

    background: white;
    color: palevioletred;
  
}"
`;

exports[`css-literal-loader multiple-components FancyHeader: multiple-components-FancyHeader.css 1`] = `
".fancy-header {

  color: red;
  width: 75px;

  ;

}
.fancy-header-variant-1 {

      background: white;
      color: palevioletred;
    
}"
`;

exports[`css-literal-loader multiple-components js : Compiled JS 1`] = `
"import { styled, css } from 'css-literal-loader/styled';
const SIZE = 75; // prettier-ignore

const FancyBox =
/*#__PURE__*/
styled('div', \\"FancyBox\\", require(\\"./multiple-components-FancyBox.css\\"), s => [s[\\"fancyBox\\"] || s[\\"fancy-box\\"], props => props.primary && (s[\\"fancyBoxVariant0\\"] || s[\\"fancy-box-variant-0\\"])]);
const FancyHeader =
/*#__PURE__*/
styled('h2', \\"FancyHeader\\", require(\\"./multiple-components-FancyHeader.css\\"), s => [s[\\"fancyHeader\\"] || s[\\"fancy-header\\"], props => props.primary && (s[\\"fancyHeaderVariant1\\"] || s[\\"fancy-header-variant-1\\"])]);"
`;

exports[`css-literal-loader no-global js : Compiled JS 1`] = `
"const SIZE = 75;
const styles = less\`
  .foo {
    color: red;
    width: \${SIZE}px;
  }

  .bar {
    composes: foo;

    font-weight: bold;
    font-style: italic;
  }
\`;"
`;

exports[`css-literal-loader styled FancyBox: styled-FancyBox.css 1`] = `
".fancy-box {

  color: red;
  width: 75px;

  @media (min-width: 420px) {
		width: 96px;
		height: 96px;
	}

  

}
.fancy-box-variant-0 {

    background: white;
    color: palevioletred;
  
}"
`;

exports[`css-literal-loader styled js : Compiled JS 1`] = `
"import { styled, css } from 'css-literal-loader/styled';
const SIZE = 75; // prettier-ignore

const FancyBox =
/*#__PURE__*/
styled('div', \\"FancyBox\\", require(\\"./styled-FancyBox.css\\"), s => [s[\\"fancyBox\\"] || s[\\"fancy-box\\"], props => props.primary && (s[\\"fancyBoxVariant0\\"] || s[\\"fancy-box-variant-0\\"])]);"
`;
